src/scanners.rs

```diff
@@ -491,6 +507,29 @@ pub fn scan_redditlink_name(data: &str) -> Option<usize> {
     }
 }
 
+pub fn scan_superscript_line(data: &str) -> usize {
+    scan_while_not(&data, is_ascii_whitespace_or_caret)
+}
+
+// scan to get matching open and closing parens
+pub fn scan_while_counting(data: &str) -> usize {
+    let mut total = 0;
+    let mut pos = 0;
+    for (i, c) in data.as_bytes().iter().enumerate() {
+        pos += 1;
+        if *c == b'(' && !is_escaped(data, i){ total += 1;}
```

Comment:
Style nit: if we're going to use a one liner here, I think we'd want to adhere to the same styling as multi-line conditionals with regarding to spacing:

    if *c == b'(' && !is_escaped(data, i) { total += 1; }

That said, I think it's probably more readable and DRYer to do something like this:

```
if !is_escaped(data, i) {
    match *c {
        b'(' => total += 1,
        b')' => total -= 1
    }
}